core/src/services/compfs/writer.rs

```diff
@@ -50,9 +56,20 @@ impl oio::Write for CompfsWriter {
     }
 
     async fn close(&mut self) -> Result<()> {
-        let f = self.file.clone();
+        let f = self
+            .file
+            .clone()
```

Comment:
Just add a `closed: bool` field, no need the `Option` as you're not taking anything out.

Reply:
`CompleteLayer` already has this check, I can remove the Option directly.